File: src/1431. Kids With the Greatest Number of Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int maxCandy = 0;
        for (auto candy : candies) {
            maxCandy = max(maxCandy, candy);
        }

        vector<bool> ans;
        for (auto candy : candies) {
            ans.push_back(candy + extraCandies >= maxCandy);
        }
        return ans;
    }
};

File: src/1431. Kids With the Greatest Number of Candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */

export function kidsWithCandies(candies, extraCandies) {
  const maxCandy = candies.reduce(
    (maxCandy, candy) => Math.max(maxCandy, candy),
    0,
  )
  return candies.map((candy) => candy + extraCandies >= maxCandy)
}
